crates/runc-rust/src/io.rs

```diff
@@ -0,0 +1,333 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+use std::fmt::{self, Debug, Formatter};
+use std::fs::File;
+use std::os::unix::io::FromRawFd;
+use std::os::unix::prelude::{AsRawFd, RawFd};
+use std::process::Command;
+use std::sync::Mutex;
+
+use nix::fcntl::OFlag;
+use nix::sys::stat::Mode;
+use nix::unistd::{Gid, Uid};
+
+pub trait RuncIO: Sync + Send {
+    /// Return write side of stdin
+    fn stdin(&self) -> Option<File> {
+        None
+    }
+
+    /// Return read side of stdout
+    fn stdout(&self) -> Option<File> {
+        None
+    }
+
+    /// Return read side of stderr
+    fn stderr(&self) -> Option<File> {
+        None
+    }
+
+    fn close(&self) {
+        unimplemented!()
+    }
+
+    /// Set IO for passed command.
+    /// Read side of stdin, write side of stdout and write side of stderr should be provided to command.
+    fn set(&self, _cmd: &mut Command) -> std::io::Result<()> {
+        unimplemented!()
+    }
+
+    // tokio version of set()
+    fn set_tk(&self, _cmd: &mut tokio::process::Command) -> std::io::Result<()> {
+        unimplemented!()
+    }
+
+    fn close_after_start(&self) {
+        unimplemented!()
+    }
```

Comment:
Not clear about these `unimplemented`. Due to time constraints?

Reply:
I' ll remove them, thank you!